corehq/apps/users/management/commands/restore_domain_membership.py

```diff
@@ -0,0 +1,236 @@
+        for change in change_feed.iter_changes(since=start_offset, forever=False):
+            if change.sequence_id > end_offset:
+                return
+
+            if change.id in seen_ids:
+                continue
+
+            seen_ids.add(change.id)
+
+            if change.deleted:
+                continue
+
+            try:
+                user = change.get_document()
+            except ResourceNotFound:
+                continue
+
+            user = CommCareUser.wrap(user)
+
+            if user_looks_ok(user):
+                continue
+
+            restore_domain_membership(user, check=check)
+
+            if change.sequence_id % 100 == 0:
+                print "Processed up to offset: {}".format(change.sequence_id)
+
+    def get_min_max_offsets(self):
+        end = get_multi_topic_offset([COMMCARE_USER])[COMMCARE_USER]
+        start = get_multi_topic_first_available_offsets([COMMCARE_USER])[COMMCARE_USER]
+        return start, end - 1  # end is next available offset
+
+
+def user_looks_ok(user):
+    if 'commcare_project' not in user.user_data:
+        return True
+
+    domain_membership = user.domain_membership
+    if any(key in domain_membership for key in extra_keys):
+        # membership has not been reset
+        return True
+
+    for key, default_value in defaults.items():
+        if domain_membership[key] != default_value:
+            # domain membership has been updated
+            return True
+
+    return False
+
+
+def restore_domain_membership(user, check=False):
+    doc_id = user._id
+    db = CommCareUser.get_db()
+    revisions = get_doc_revisions(db, doc_id)
+    for rev in revisions[1:]:
+        doc = get_doc_rev(db, doc_id, rev)
+        if not doc:
+            continue
+        prev_user = CommCareUser.wrap(doc)
+        if user_looks_ok(prev_user):
+            if user.location_id != prev_user.domain_membership.location_id:
+                continue
+
+            if user.assigned_location_ids != prev_user.domain_membership.assigned_location_ids:
+                continue
+
+            if check:
+                print 'Ready to patch user: {} ({})'.format(user.domain, doc_id)
+                old = json.dumps(user.domain_membership.to_json(), indent=2)
+                print 'Old domain membership: \n{}\n'.format(old)
+                new = json.dumps(prev_user.domain_membership.to_json(), indent=2)
+                print 'New domain membership: \n{}\n'.format(new)
+                if not confirm('Proceed with updating user?'):
+                    return
+
+            print "Patching user: {} ({})".format(user.domain, doc_id)
+            prev_domain_membership = prev_user.domain_membership
+            user.domain_membership = prev_domain_membership
+            user.save()
+            return
+
+    print 'Unable to fix user: {} ({})'.format(user.domain, doc_id)
+
+
+def get_doc_revisions(db, doc_id):
+    res = db.get(doc_id, revs=True)
```

Comment:
@snopoke Is this management command still in use?  We're trying to move couchdbkit off of restkit to the [cloudant API](https://python-cloudant.readthedocs.io/en/latest/document.html#cloudant.document.Document), but the cloudant API doesn't support the ```revs``` kwarg.

Reply:
no, it was a one time command - fine to remove